test/models/user_test.rb

```diff
@@ -874,4 +876,24 @@ def editing_self_helper
     user.timezone = ''
     assert user.valid?
   end
+
+  test "changing user password as admin without setting current password" do
+    user = FactoryGirl.create(:user)
+    as_admin do
+      user.password = "newpassword"
+      assert user.save
+    end
+    assert user.valid?
+  end
+
+  test "changing user's own password with incorrect current password" do
+    user = FactoryGirl.create(:user)
+    as_user user do
+      user.password = "newpassword"
+      user.current_password = "hatatitla"
+    end
+    assert user.save
```

Comment:
This test doesn't seem to check what's described - it says it checks the user changing their own password with an incorrect current password, but this line and lines 896/7 actually suggest it _passes_. It should be _failing_ validation.

I think it's because the save/validation are not in the `as_user user` block.